src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -234,18 +236,73 @@ public Declaration.Scoped createRecord(Cursor c, Declaration.Scoped.Kind scopeKi
         }
     }
 
+    final Type.Declared recordDeclaration(Cursor parent, Cursor recordCursor) {
+        List<Declaration> pendingFields = new ArrayList<>();
+        List<Variable> pendingBitFields = new ArrayList<>();
+        AtomicReference<Position> pendingBitfieldsPos = new AtomicReference<>();
+        recordCursor.forEach(fc -> {
+            if (Utils.isFlattenable(fc)) {
+                if (fc.isBitField()) {
+                    if (pendingBitfieldsPos.get() == null) {
+                        pendingBitfieldsPos.set(CursorPosition.of(fc));
+                    }
+                    Type fieldType = typeMaker.makeType(fc.type());
+                    Variable bitfieldDecl = Declaration.bitfield(CursorPosition.of(fc), fc.spelling(), fc.getBitFieldWidth(), fieldType);
+                    if (!fc.spelling().isEmpty()) {
+                        ClangOffsetOf.with(bitfieldDecl, parent.type().getOffsetOf(fc.spelling()));
+                    }
+                    pendingBitFields.add(bitfieldDecl);
+                } else {
+                    if (!pendingBitFields.isEmpty()) {
+                        pendingFields.add(Declaration.bitfields(pendingBitfieldsPos.get(), pendingBitFields.toArray(Variable[]::new)));
+                        pendingBitFields.clear();
+                        pendingBitfieldsPos.set(null);
+                    }
+                    if (fc.isAnonymousStruct()) {
+                        // process struct recursively
+                        pendingFields.add(recordDeclaration(parent, fc).tree());
+                    } else if (!fc.isBitField() && !fc.spelling().isEmpty()) {
```

Comment:
The condition here seems to be unnecessary? We should already be in the no-bitfield case due to enclosing if/else, and the isAnonymousStruct check should filter out fields without a name?